code/game/objects/items/stacks/stack.dm

```diff
@@ -341,6 +344,28 @@
 	else
 		return ..()
 
+//Verb to split stacks
+/obj/item/stack/verb/split_verb()
+	set src in view()
+	set name = "Split"
+
+
+	if (!Adjacent(usr))
+		usr << SPAN_WARNING("You need to be in arm's reach for that!")
+		return
+
+	var/quantity = input(usr,
+	"This stack contains [amount]/[max_amount]. How many would you like to split off into a new stack?\n\
+	The new stack will be put into your hands if possible", "Split Stack", round(amount * 0.5)) as num
+
```

Comment:
Safety checks missed after quantity input.

Reply:
safety checks are in the split function iirc. 
What kind of checks would you like therE?